tests/ert_tests/ert3/data/test_transformations.py

```diff
@@ -165,3 +245,39 @@ async def test_transform_output_sequence(tmpdir):
     for record, data in zip(records, test_data):
         assert isinstance(record, NumericalRecord)
         assert data == record.data
+
+
+@pytest.mark.asyncio
+async def test_eclsum_transformation(tmp_path):
+    _create_synthetic_smry(tmp_path, length=3)
+    numrecordtree = await EclSumTransformation(smry_keys=["FOPT"]).transform_output(
+        location=tmp_path / "TEST", mime=None
+    )
+    data = numrecordtree.flat_record_dict["FOPT"].data
```

Comment:
Here an option would be to fetch data via `numrecordtree.data["FOPT"].data` ie. using the `data` property but that would mean extending
```
record_data = Union[
    numerical_record_data,
    blob_record_data,
]
``` hmm :thinking: For sure not part this PR though just food for thought ...